File: rustbridge-java/rustbridge-core/src/main/java/com/rustbridge/LogCallback.java
package com.rustbridge;

import org.jetbrains.annotations.NotNull;

/**
 * Callback interface for receiving log messages from plugins.
 */
@FunctionalInterface
public interface LogCallback {
    /**
     * Called when the plugin emits a log message.
     *
     * @param level   the log level
     * @param target  the log target (usually module path)
     * @param message the log message
     */
    void log(@NotNull LogLevel level, @NotNull String target, @NotNull String message);
}

File: rustbridge-csharp/RustBridge.Core/LogCallback.cs
namespace RustBridge;

/// <summary>
/// Delegate for receiving log messages from plugins.
/// </summary>
/// <param name="level">The log level.</param>
/// <param name="target">The log target (usually module path).</param>
/// <param name="message">The log message.</param>
public delegate void LogCallback(LogLevel level, string target, string message);
